src/prpy/planning/logged.py

```diff
@@ -0,0 +1,145 @@
+    """Planner wrapper which serializes planning requets and responses.
+
+    LoggedPlanner delegates all planning calls to the sub-planner
+    passed to it on construction.  It does not affect the input to
+    or output from this planner.
+
+    On every request, LoggedPlanner serializes the environment,
+    planning method request arguments, and the returned trajectory.
+    It creates a timestamped yaml file in the current directory:
+     - log-20160101-010203.567890.yaml
+    """
+
+    def __init__(self, planner):
+        super(LoggedPlanner, self).__init__()
+        self.planner = planner
+    
+    def __str__(self):
+        return 'Logged({0:s})'.format(self.planner)
+    
+    def has_planning_method(self, method_name):
+        return self.planner.has_planning_method(method_name)
+    
+    def get_planning_method_names(self):
+        return self.planner.get_planning_method_names()
+    
+    def get_planners(self, method_name):
+        return [self.planner]
+    
+    def plan(self, method, args, kw_args):
+        
+        # get log file name
+        stamp = time.time()
+        struct = time.localtime(stamp)
+        fn = 'log-{}.{:06d}.yaml'.format(
+            time.strftime('%Y%m%d-%H%M%S', struct),
+            int(1.0e6*(stamp-math.floor(stamp))))
+        
+        # retrieve enviroment
+        if 'robot' in kw_args:
+            env = kw_args['robot'].GetEnv()
+        elif len(args) > 0:
+            env = args[0].GetEnv()
+        else:
+            raise RuntimeError('could not retrieve env from planning request!')
+        
+        # serialize environment
+        envdict = prpy.serialization.serialize_environment(env, uri_only=True)
+        
+        # serialize planning request
+        reqdict = {}
+        reqdict['method'] = method # string
+        reqdict['args'] = []
+        for arg in args:
+            reqdict['args'].append(serialize_or_errorstr(arg))
+        reqdict['kw_args'] = {}
+        for key,value in kw_args.items():
+            # if key serialization fails, it thros UnsupportedTypeSerializationException
+            key = prpy.serialization.serialize(key)
+            reqdict['kw_args'][key] = serialize_or_errorstr(value)
+        
+        # get ready to serialize result
+        resdict = {}
+        
+        try:
+            plan_fn = getattr(self.planner, method)
+            traj = plan_fn(*args, **kw_args)
+            resdict['ok'] = True
+            resdict['traj_first'] = list(map(float,traj.GetWaypoint(0)))
+            resdict['traj_last'] = list(map(float,traj.GetWaypoint(traj.GetNumWaypoints()-1)))
+            resdict['traj'] = traj.serialize(0)
+            return traj
+            
+        except prpy.planning.PlanningError as ex:
+            resdict['ok'] = False
+            resdict['exception'] = str(ex)
+            raise
+        
+        except:
+            resdict['ok'] = False
+            resdict['exception'] = 'unknown exception type'
```

Comment:
I did the former, but I prefer if non-expected exceptions have the additional "Unknown exception type" text.